yarn-project/acir-simulator/src/client/view_data_oracle.ts

```diff
@@ -35,6 +35,71 @@ export class ViewDataOracle extends TypedOracle {
     return this.db.getSecretKey(this.contractAddress, owner);
   }
 
+  /**
+   * Fetches the index and sibling path of a leaf at a given block from a given tree.
+   * @param blockNumber - The block number at which to get the membership witness.
+   * @param treeId - Id of the tree to get the sibling path from.
+   * @param leafValue - The leaf value
+   * @returns The index and sibling path concatenated [index, sibling_path]
+   */
+  public async getMembershipWitness(blockNumber: number, treeId: MerkleTreeId, leafValue: Fr): Promise<Fr[]> {
+    const index = await this.db.findLeafIndex(blockNumber, treeId, leafValue);
+    if (!index) {
+      throw new Error(`Leaf value: ${leafValue} not found in tree ${treeId}`);
+    }
+    const siblingPath = await this.db.getSiblingPath(blockNumber, treeId, index);
+    return [new Fr(index), ...siblingPath];
+  }
+
+  /**
+   * Fetches a sibling path at a given block and index from a tree specified by `treeId`.
+   * @param blockNumber - The block number at which to get the membership witness.
+   * @param treeId - Id of the tree to get the sibling path from.
+   * @param index - Index of the leaf to get sibling path for
+   * @returns The index and sibling path concatenated [index, sibling_path]
+   */
+  public getSiblingPath(blockNumber: number, treeId: MerkleTreeId, index: Fr): Promise<Fr[]> {
+    return this.db.getSiblingPath(blockNumber, treeId, index.toBigInt());
+  }
+
+  /**
+   * Returns a low nullifier membership witness for a given nullifier at a given block.
```

Comment:
+1, not sure if i argued against you in the past lol

Reply:
"Predecessor" and "successor" implies that the leaves are somehow inextricably connected. But in fact these relationships can change as new leaves are added to the tree. My own predecessors and successors are fixed people; I can't change them if a new person comes into my life (although I guess I could start a new family... but that wouldn't break the link I have with my current successors).

 "Indexed Merkle Tree" is literally the name coined by the people who invented the tree https://eprint.iacr.org/2021/1263.pdf. Seems confusing to try to coin a new term.
 
 It's basically a linked list. Is there terminology that can be borrowed from linked list literature?
 
 I like "pointer" because it points, and pointers can be updated.